### ISSUE TITLE
How does ignore_empty work?

### ISSUE THREAD

**Issue by bobbwal** [getgrav/grav#2417]
I'm trying to set page permissions from a blueprint 
```
header.access.site:
  type: checkboxes
  label: Permissions
  data-options@: '\Grav\Common\User\Group::groupNames'
  use: keys
  validate:
    type: array
```
Output:
```
access:
  site:
    level01: true
    level02: false
    level03: false
```
The problem is the false entries seem to break the permissions, without these it works fine so I need try and prevent them being added. I've tried using `ignore_empty` hoping it would prevent empty checkboxes from creating the false entries but it throws an error:

![ignore_empty](https://user-images.githubusercontent.com/391847/54723826-c83e9900-4b60-11e9-8a94-8bc7c0c1ffd1.jpg)

Any ideas?

**Comment by bobbwal** [comment]
Any ideas on this?

**Comment by mahagr** [comment]
@bobbwal I'm not able to reproduce the issue; it works just fine for me in admin.

**Comment by bobbwal** [comment]
@mahagr thanks for the reply, much appreciated. I guess I have something else going on then. When working correctly would `ignore_empty` remove the false entries listed in my output above?

**Comment by mahagr** [comment]
No, they won't because the items are either true or false, not null or empty string.

But I couldn't get the code to fail hard as you did. What is the easiest way to reproduce the issue?

**Comment by bobbwal** [comment]
Ah okay, this isn't going to solve my problem then but I have another option to explore. Regarding the error, I've just tried on a fresh install and I get the same invalid argument. The code below should recreate the error:
`user\config\groups.yaml`
```
registered:
  icon: users
  readableName: 'Registered Users'
  description: 'The group of registered users'
  access:
    site:
      login: true
paid:
  readableName: 'Paid Members'
  description: 'The group of paid members'
  icon: money
  access:
    site:
      login: true
      paid: true
```

`user\themes\quark\templates\add_access.html.twig`
```
{% extends 'partials/base.html.twig' %}

{% block content %}
    {{ page.content|raw }}
{% endblock %}
```
`user\themes\quark\blueprints\add_access.yaml`
```
title: PLUGIN_ADMIN.DEFAULT
'@extends':
  type: default
  context: blueprints://pages

form:
  fields:
    tabs:
      fields:
        options:
          fields:
            publishing:
              fields:
                header.access.site:
                  ordering@: 2
                  type: checkboxes
                  toggleable: true
                  label: Groups
                  help: Groups
                  default: false
                  ignore_empty: true
                  data-options@: '\Grav\Common\User\Group::groupNames'
                  use: keys
                  validate:
                    type: array
```

**Comment by mahagr** [comment]
I tested the code and it does not break the site when using Grav 1.6 RC, though ignoring empty doesn't work because of false isn't considered to be empty. 

Can you try latest 1.6 RC if it fixes your issues? We're about to release it so I am not sure if we're backporting the fix to 1.5.

**Comment by bobbwal** [comment]
No worries, it's working fine with 1.6.0-rc.4

**Comment by mahagr** [comment]
Ok, marked it as fixed in 1.6
